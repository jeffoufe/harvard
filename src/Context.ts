import React, { SetStateAction, Dispatch } from "react";

interface Context {
    isMenuOpen: boolean,
    setMenuOpen: Dispatch<SetStateAction<boolean>>
}

export default React.createContext<Context>({} as Context)