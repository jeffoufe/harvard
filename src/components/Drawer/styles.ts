import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

const drawerWidth = 240;

export const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        drawer: {
            width: drawerWidth,
            flexShrink: 0,
        },
        drawerPaper: {
            width: drawerWidth,
        },
        drawerHeader: {
            display: 'flex',
            alignItems: 'center',
            padding: theme.spacing(0, 1),
            ...theme.mixins.toolbar,
            justifyContent: 'flex-end',
        },
    })
)