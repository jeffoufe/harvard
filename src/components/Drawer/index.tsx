import React, { useContext } from 'react';
import { Drawer, IconButton, Divider, List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import BrushIcon from '@material-ui/icons/Brush';
import useStyles from './styles';
import Context from '../../Context';

const LeftDrawer = () => {
    const classes = useStyles();
    const { isMenuOpen, setMenuOpen } = useContext(Context);
    const handleDrawerClose = () => setMenuOpen(false);

    return (
        <Drawer
            variant="persistent"
            anchor="left"
            open={isMenuOpen}
            classes={{
                paper: classes.drawerPaper,
            }}
        >
            <div>
                <IconButton onClick={handleDrawerClose} className={classes.homeButton}>
                    <ChevronLeftIcon />
                    Harvard
                </IconButton>
            </div>
            <Divider />
            <List>
                <ListItem>
                        <ListItemIcon>
                            <BrushIcon />
                        </ListItemIcon>
                        <ListItemText primary='Prints' />
                </ListItem>
            </List>
        </Drawer>
    )
}

LeftDrawer.displayName = 'Drawer';
export default LeftDrawer;