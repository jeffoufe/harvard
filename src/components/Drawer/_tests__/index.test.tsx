import React from 'react';
import Drawer from '..';
import Context from '../../../Context';
import { createMount } from '@material-ui/core/test-utils';

describe('Drawer', () => {
    let mount;
    let wrapper;
    const closeDrawerHandler = jest.fn()

    beforeAll(() => {
        mount = createMount();
    });
    
    afterAll(() => {
        mount.cleanUp();
    });

    it('render correctly', () => {
        wrapper = mount(
            <Context.Provider value={[true, closeDrawerHandler]}>
                <Drawer />
            </Context.Provider>
        )

        expect(wrapper).toMatchSnapshot();
    })

    it('props are passed', () => {
       expect(wrapper.find('ForwardRef(Drawer)').props()['open']).toBeTruthy();
    })

    it('handlers', () => {
        wrapper.find('ForwardRef(IconButton)').props()['onClick']();
        expect(closeDrawerHandler).toHaveBeenCalledWith(false);
    })
})