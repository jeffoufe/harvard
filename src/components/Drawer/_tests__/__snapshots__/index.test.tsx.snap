// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer render correctly 1`] = `
<Drawer>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-2",
      }
    }
    open={true}
    variant="persistent"
  >
    <ForwardRef(Drawer)
      anchor="left"
      classes={
        Object {
          "docked": "MuiDrawer-docked",
          "modal": "MuiDrawer-modal",
          "paper": "MuiDrawer-paper makeStyles-drawerPaper-2",
          "paperAnchorBottom": "MuiDrawer-paperAnchorBottom",
          "paperAnchorDockedBottom": "MuiDrawer-paperAnchorDockedBottom",
          "paperAnchorDockedLeft": "MuiDrawer-paperAnchorDockedLeft",
          "paperAnchorDockedRight": "MuiDrawer-paperAnchorDockedRight",
          "paperAnchorDockedTop": "MuiDrawer-paperAnchorDockedTop",
          "paperAnchorLeft": "MuiDrawer-paperAnchorLeft",
          "paperAnchorRight": "MuiDrawer-paperAnchorRight",
          "paperAnchorTop": "MuiDrawer-paperAnchorTop",
          "root": "MuiDrawer-root",
        }
      }
      open={true}
      variant="persistent"
    >
      <div
        className="MuiDrawer-root MuiDrawer-docked"
      >
        <ForwardRef(Slide)
          appear={false}
          direction="right"
          in={true}
          timeout={
            Object {
              "enter": 225,
              "exit": 195,
            }
          }
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            nodeRef={
              Object {
                "current": <div
                  class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
                >
                  <div>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                        Harvard
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <hr
                    class="MuiDivider-root"
                  />
                  <ul
                    class="MuiList-root MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters"
                    >
                      <div
                        class="MuiListItemIcon-root"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 14c-1.66 0-3 1.34-3 3 0 1.31-1.16 2-2 2 .92 1.22 2.49 2 4 2 2.21 0 4-1.79 4-4 0-1.66-1.34-3-3-3zm13.71-9.37l-1.34-1.34a.9959.9959 0 00-1.41 0L9 12.25 11.75 15l8.96-8.96c.39-.39.39-1.02 0-1.41z"
                          />
                        </svg>
                      </div>
                      <div
                        class="MuiListItemText-root"
                      >
                        <span
                          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                        >
                          Prints
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>,
              }
            }
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={
              Object {
                "enter": 225,
                "exit": 195,
              }
            }
            unmountOnExit={false}
          >
            <WithStyles(ForwardRef(Paper))
              className="MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
              elevation={0}
              square={true}
              style={
                Object {
                  "visibility": undefined,
                }
              }
            >
              <ForwardRef(Paper)
                className="MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
                classes={
                  Object {
                    "elevation0": "MuiPaper-elevation0",
                    "elevation1": "MuiPaper-elevation1",
                    "elevation10": "MuiPaper-elevation10",
                    "elevation11": "MuiPaper-elevation11",
                    "elevation12": "MuiPaper-elevation12",
                    "elevation13": "MuiPaper-elevation13",
                    "elevation14": "MuiPaper-elevation14",
                    "elevation15": "MuiPaper-elevation15",
                    "elevation16": "MuiPaper-elevation16",
                    "elevation17": "MuiPaper-elevation17",
                    "elevation18": "MuiPaper-elevation18",
                    "elevation19": "MuiPaper-elevation19",
                    "elevation2": "MuiPaper-elevation2",
                    "elevation20": "MuiPaper-elevation20",
                    "elevation21": "MuiPaper-elevation21",
                    "elevation22": "MuiPaper-elevation22",
                    "elevation23": "MuiPaper-elevation23",
                    "elevation24": "MuiPaper-elevation24",
                    "elevation3": "MuiPaper-elevation3",
                    "elevation4": "MuiPaper-elevation4",
                    "elevation5": "MuiPaper-elevation5",
                    "elevation6": "MuiPaper-elevation6",
                    "elevation7": "MuiPaper-elevation7",
                    "elevation8": "MuiPaper-elevation8",
                    "elevation9": "MuiPaper-elevation9",
                    "outlined": "MuiPaper-outlined",
                    "root": "MuiPaper-root",
                    "rounded": "MuiPaper-rounded",
                  }
                }
                elevation={0}
                square={true}
                style={
                  Object {
                    "visibility": undefined,
                  }
                }
              >
                <div
                  className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
                  style={
                    Object {
                      "visibility": undefined,
                    }
                  }
                >
                  <div>
                    <WithStyles(ForwardRef(IconButton))
                      onClick={[Function]}
                    >
                      <ForwardRef(IconButton)
                        classes={
                          Object {
                            "colorInherit": "MuiIconButton-colorInherit",
                            "colorPrimary": "MuiIconButton-colorPrimary",
                            "colorSecondary": "MuiIconButton-colorSecondary",
                            "disabled": "Mui-disabled",
                            "edgeEnd": "MuiIconButton-edgeEnd",
                            "edgeStart": "MuiIconButton-edgeStart",
                            "label": "MuiIconButton-label",
                            "root": "MuiIconButton-root",
                            "sizeSmall": "MuiIconButton-sizeSmall",
                          }
                        }
                        onClick={[Function]}
                      >
                        <WithStyles(ForwardRef(ButtonBase))
                          centerRipple={true}
                          className="MuiIconButton-root"
                          disabled={false}
                          focusRipple={true}
                          onClick={[Function]}
                        >
                          <ForwardRef(ButtonBase)
                            centerRipple={true}
                            className="MuiIconButton-root"
                            classes={
                              Object {
                                "disabled": "Mui-disabled",
                                "focusVisible": "Mui-focusVisible",
                                "root": "MuiButtonBase-root",
                              }
                            }
                            disabled={false}
                            focusRipple={true}
                            onClick={[Function]}
                          >
                            <button
                              className="MuiButtonBase-root MuiIconButton-root"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label"
                              >
                                <ForwardRef>
                                  <WithStyles(ForwardRef(SvgIcon))>
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                                Harvard
                              </span>
                              <WithStyles(memo)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(memo)>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                  </div>
                  <WithStyles(ForwardRef(Divider))>
                    <ForwardRef(Divider)
                      classes={
                        Object {
                          "absolute": "MuiDivider-absolute",
                          "flexItem": "MuiDivider-flexItem",
                          "inset": "MuiDivider-inset",
                          "light": "MuiDivider-light",
                          "middle": "MuiDivider-middle",
                          "root": "MuiDivider-root",
                          "vertical": "MuiDivider-vertical",
                        }
                      }
                    >
                      <hr
                        className="MuiDivider-root"
                      />
                    </ForwardRef(Divider)>
                  </WithStyles(ForwardRef(Divider))>
                  <WithStyles(ForwardRef(List))>
                    <ForwardRef(List)
                      classes={
                        Object {
                          "dense": "MuiList-dense",
                          "padding": "MuiList-padding",
                          "root": "MuiList-root",
                          "subheader": "MuiList-subheader",
                        }
                      }
                    >
                      <ul
                        className="MuiList-root MuiList-padding"
                      >
                        <WithStyles(ForwardRef(ListItem))>
                          <ForwardRef(ListItem)
                            classes={
                              Object {
                                "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                                "button": "MuiListItem-button",
                                "container": "MuiListItem-container",
                                "dense": "MuiListItem-dense",
                                "disabled": "Mui-disabled",
                                "divider": "MuiListItem-divider",
                                "focusVisible": "Mui-focusVisible",
                                "gutters": "MuiListItem-gutters",
                                "root": "MuiListItem-root",
                                "secondaryAction": "MuiListItem-secondaryAction",
                                "selected": "Mui-selected",
                              }
                            }
                          >
                            <li
                              className="MuiListItem-root MuiListItem-gutters"
                              disabled={false}
                            >
                              <WithStyles(ForwardRef(ListItemIcon))
                                key=".0"
                              >
                                <ForwardRef(ListItemIcon)
                                  classes={
                                    Object {
                                      "alignItemsFlexStart": "MuiListItemIcon-alignItemsFlexStart",
                                      "root": "MuiListItemIcon-root",
                                    }
                                  }
                                >
                                  <div
                                    className="MuiListItemIcon-root"
                                  >
                                    <ForwardRef>
                                      <WithStyles(ForwardRef(SvgIcon))>
                                        <ForwardRef(SvgIcon)
                                          classes={
                                            Object {
                                              "colorAction": "MuiSvgIcon-colorAction",
                                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                                              "colorError": "MuiSvgIcon-colorError",
                                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                              "root": "MuiSvgIcon-root",
                                            }
                                          }
                                        >
                                          <svg
                                            aria-hidden={true}
                                            className="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <path
                                              d="M7 14c-1.66 0-3 1.34-3 3 0 1.31-1.16 2-2 2 .92 1.22 2.49 2 4 2 2.21 0 4-1.79 4-4 0-1.66-1.34-3-3-3zm13.71-9.37l-1.34-1.34a.9959.9959 0 00-1.41 0L9 12.25 11.75 15l8.96-8.96c.39-.39.39-1.02 0-1.41z"
                                            />
                                          </svg>
                                        </ForwardRef(SvgIcon)>
                                      </WithStyles(ForwardRef(SvgIcon))>
                                    </ForwardRef>
                                  </div>
                                </ForwardRef(ListItemIcon)>
                              </WithStyles(ForwardRef(ListItemIcon))>
                              <WithStyles(ForwardRef(ListItemText))
                                key=".1"
                                primary="Prints"
                              >
                                <ForwardRef(ListItemText)
                                  classes={
                                    Object {
                                      "dense": "MuiListItemText-dense",
                                      "inset": "MuiListItemText-inset",
                                      "multiline": "MuiListItemText-multiline",
                                      "primary": "MuiListItemText-primary",
                                      "root": "MuiListItemText-root",
                                      "secondary": "MuiListItemText-secondary",
                                    }
                                  }
                                  primary="Prints"
                                >
                                  <div
                                    className="MuiListItemText-root"
                                  >
                                    <WithStyles(ForwardRef(Typography))
                                      className="MuiListItemText-primary"
                                      component="span"
                                      display="block"
                                      variant="body1"
                                    >
                                      <ForwardRef(Typography)
                                        className="MuiListItemText-primary"
                                        classes={
                                          Object {
                                            "alignCenter": "MuiTypography-alignCenter",
                                            "alignJustify": "MuiTypography-alignJustify",
                                            "alignLeft": "MuiTypography-alignLeft",
                                            "alignRight": "MuiTypography-alignRight",
                                            "body1": "MuiTypography-body1",
                                            "body2": "MuiTypography-body2",
                                            "button": "MuiTypography-button",
                                            "caption": "MuiTypography-caption",
                                            "colorError": "MuiTypography-colorError",
                                            "colorInherit": "MuiTypography-colorInherit",
                                            "colorPrimary": "MuiTypography-colorPrimary",
                                            "colorSecondary": "MuiTypography-colorSecondary",
                                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                            "displayBlock": "MuiTypography-displayBlock",
                                            "displayInline": "MuiTypography-displayInline",
                                            "gutterBottom": "MuiTypography-gutterBottom",
                                            "h1": "MuiTypography-h1",
                                            "h2": "MuiTypography-h2",
                                            "h3": "MuiTypography-h3",
                                            "h4": "MuiTypography-h4",
                                            "h5": "MuiTypography-h5",
                                            "h6": "MuiTypography-h6",
                                            "noWrap": "MuiTypography-noWrap",
                                            "overline": "MuiTypography-overline",
                                            "paragraph": "MuiTypography-paragraph",
                                            "root": "MuiTypography-root",
                                            "srOnly": "MuiTypography-srOnly",
                                            "subtitle1": "MuiTypography-subtitle1",
                                            "subtitle2": "MuiTypography-subtitle2",
                                          }
                                        }
                                        component="span"
                                        display="block"
                                        variant="body1"
                                      >
                                        <span
                                          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                        >
                                          Prints
                                        </span>
                                      </ForwardRef(Typography)>
                                    </WithStyles(ForwardRef(Typography))>
                                  </div>
                                </ForwardRef(ListItemText)>
                              </WithStyles(ForwardRef(ListItemText))>
                            </li>
                          </ForwardRef(ListItem)>
                        </WithStyles(ForwardRef(ListItem))>
                      </ul>
                    </ForwardRef(List)>
                  </WithStyles(ForwardRef(List))>
                </div>
              </ForwardRef(Paper)>
            </WithStyles(ForwardRef(Paper))>
          </Transition>
        </ForwardRef(Slide)>
      </div>
    </ForwardRef(Drawer)>
  </WithStyles(ForwardRef(Drawer))>
</Drawer>
`;
